includes/class-wc-ajax.php

```diff
@@ -1306,6 +1306,13 @@ public static function product_ordering() {
 			$index ++;
 			$menu_orders[ $id ] = $index;
 			$wpdb->update( $wpdb->posts, array( 'menu_order' => $index ), array( 'ID' => $id ) );
+			
+			/**
+			 * When a single product has gotten it's ordering updated.
+			 * $id The product ID
+			 * $index The new menu order
+			*/
+			do_action( 'woocommerce_after_single_product_ordering', $id, $index );
```

Comment:
So I haven't had time to investigate that one, and you're sure that it would fire after an Ajax sorting?

Digging up these actions in seeming unrelated placed is also something that is tough for your users. Maybe another argument for why this hook is more intuitive.